esphome/components/toshiba/toshiba.cpp

```diff
@@ -97,6 +99,300 @@ const std::vector<uint8_t> RAC_PT1411HWRU_TEMPERATURE_F{0x10, 0x30, 0x00, 0x20,
+// - Check bytes 2-3: if 0x1FE0 → AUTO/DRY/low-temp-COOL (distinguish by temp code)
+// - Otherwise: COOL/HEAT/FAN_ONLY (distinguish by temp code and byte 5)
+
+/**
+ * Get fan speed encoding for RAS-2819T first packet (rc_code_1, bytes 2-3)
+ */
+uint16_t get_ras_2819t_fan_code(climate::ClimateFanMode fan_mode) {
+  switch (fan_mode) {
+    case climate::CLIMATE_FAN_QUIET:
+      return RAS_2819T_FAN_QUIET;
+    case climate::CLIMATE_FAN_LOW:
+      return RAS_2819T_FAN_LOW;
+    case climate::CLIMATE_FAN_MEDIUM:
+      return RAS_2819T_FAN_MEDIUM;
+    case climate::CLIMATE_FAN_HIGH:
+      return RAS_2819T_FAN_HIGH;
+    case climate::CLIMATE_FAN_AUTO:
+    default:
+      return RAS_2819T_FAN_AUTO;
+  }
+}
+
+/**
+ * Get fan speed encoding for RAS-2819T rc_code_2 packet (second packet)
+ */
+struct Ras2819tSecondPacketCodes {
+  uint8_t fan_byte;
+  Ras2819tPacketSuffix suffix;
+};
+
+Ras2819tSecondPacketCodes get_ras_2819t_second_packet_codes(climate::ClimateFanMode fan_mode) {
+  switch (fan_mode) {
+    case climate::CLIMATE_FAN_QUIET:
+      return {RAS_2819T_FAN2_QUIET, RAS_2819T_SUFFIX_QUIET};
+    case climate::CLIMATE_FAN_LOW:
+      return {RAS_2819T_FAN2_LOW, RAS_2819T_SUFFIX_LOW};
+    case climate::CLIMATE_FAN_MEDIUM:
+      return {RAS_2819T_FAN2_MEDIUM, RAS_2819T_SUFFIX_MEDIUM};
+    case climate::CLIMATE_FAN_HIGH:
+      return {RAS_2819T_FAN2_HIGH, RAS_2819T_SUFFIX_HIGH};
+    case climate::CLIMATE_FAN_AUTO:
+    default:
+      return {RAS_2819T_FAN2_AUTO, RAS_2819T_SUFFIX_AUTO};
+  }
+}
+
+/**
+ * Get temperature code for RAS-2819T protocol
+ */
+uint8_t get_ras_2819t_temp_code(float temperature) {
+  // Temperature codes for RAS-2819T protocol (18-30°C)
+  static const uint8_t RAS_2819T_TEMP_CODES[] = {
+      0x10,  // 18°C
+      0x30,  // 19°C
+      0x20,  // 20°C
+      0x60,  // 21°C
+      0x70,  // 22°C
+      0x50,  // 23°C
+      0x40,  // 24°C
+      0xC0,  // 25°C
+      0xD0,  // 26°C
+      0x90,  // 27°C
+      0x80,  // 28°C
+      0xA0,  // 29°C
+      0xB0   // 30°C
+  };
+
+  int temp_index = static_cast<int>(temperature) - 18;
+  if (temp_index < 0 || temp_index >= static_cast<int>(sizeof(RAS_2819T_TEMP_CODES))) {
+    ESP_LOGW(TAG, "Temperature %.1f°C out of range [18-30°C], defaulting to 24°C", temperature);
+    return 0x40;  // Default to 24°C
+  }
+
+  return RAS_2819T_TEMP_CODES[temp_index];
+}
+
+/**
+ * Decode temperature from RAS-2819T temp code
+ */
+float decode_ras_2819t_temperature(uint8_t temp_code) {
+  // Map temp codes back to temperatures using array lookup for better performance
+  static const uint8_t RAS_2819T_TEMP_CODES_DECODE[] = {0x10, 0x30, 0x20, 0x60, 0x70, 0x50, 0x40,
+                                                        0xC0, 0xD0, 0x90, 0x80, 0xA0, 0xB0};
+
+  uint8_t base_temp_code = temp_code & 0xF0;
+
+  // Linear search through the small array (faster than map for small datasets)
+  for (size_t i = 0; i < sizeof(RAS_2819T_TEMP_CODES_DECODE); i++) {
```

Comment:
No need for a 32-bit `int` to count to 13. 🙂
```suggestion
  for (uint8_t i = 0; i < sizeof(RAS_2819T_TEMP_CODES_DECODE); i++) {
```